kernel/kernel-api/src/test/scala/io/delta/kernel/internal/DeltaLogActionUtilsSuite.scala

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal
+
+import java.io.FileNotFoundException
+
+import scala.collection.JavaConverters._
+import scala.reflect.ClassTag
+
+import io.delta.kernel.exceptions.{InvalidTableException, KernelException, TableNotFoundException}
+import io.delta.kernel.internal.util.FileNames
+import io.delta.kernel.utils.FileStatus
+import org.scalatest.funsuite.AnyFunSuite
+import io.delta.kernel.internal.DeltaLogActionUtils.{getCommitFilesForVersionRange, verifyDeltaVersions}
+import io.delta.kernel.test.MockFileSystemClientUtils
+
+class DeltaLogActionUtilsSuite extends AnyFunSuite with MockFileSystemClientUtils {
+
+  //////////////////////////////////////////////////////////////////////////////////
+  // verifyDeltaVersions tests
+  //////////////////////////////////////////////////////////////////////////////////
+
+  def getCommitFiles(versions: Seq[Long]): java.util.List[FileStatus] = {
+    versions
+      .map(v => FileStatus.of(FileNames.deltaFile(logPath, v), 0, 0))
+      .asJava
+  }
+
+  test("verifyDeltaVersions") {
+    // Basic correct use case
+    verifyDeltaVersions(
+      getCommitFiles(Seq(1, 2, 3)),
+      1,
+      3,
+      dataPath
+    )
+    // Only one version provided
+    verifyDeltaVersions(
+      getCommitFiles(Seq(1)),
+      1,
+      1,
+      dataPath
+    )
+    // Non-continuous versions
+    intercept[InvalidTableException] {
+      verifyDeltaVersions(
+        getCommitFiles(Seq(1, 3, 4)),
+        1,
+        4,
+        dataPath
+      )
+    }
+    // End-version or start-version not right
+    intercept[KernelException] {
+      verifyDeltaVersions(
+        getCommitFiles(Seq(1, 2, 3)),
+        0,
+        3,
+        dataPath
+      )
+    }
+    intercept[KernelException] {
+      verifyDeltaVersions(
+        getCommitFiles(Seq(1, 2, 3)),
+        1,
+        4,
+        dataPath
+      )
+    }
+    // Empty versions
+    intercept[KernelException] {
+      verifyDeltaVersions(
+        getCommitFiles(Seq()),
+        1,
+        4,
+        dataPath
+      )
+    }
+    // Unsorted or duplicates (shouldn't be possible)
+    intercept[InvalidTableException] {
+      verifyDeltaVersions(
+        getCommitFiles(Seq(1, 1, 2)),
+        1,
+        4,
```

Comment:
should this be 2? to simplify exctly what this test case is testing?

Reply:
yes good catch